test/integration/xds_config_tracker_integration_test.cc

```diff
@@ -0,0 +1,291 @@
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "absl/synchronization/mutex.h"
+#include "gtest/gtest.h"
+
+using testing::HasSubstr;
+
+namespace Envoy {
+namespace {
+
+constexpr char XDS_CLUSTER_NAME[] = "xds_cluster";
+
+class TestXdsConfigTracker : public Config::XdsConfigTracker {
+public:
+  TestXdsConfigTracker() {
+    ReceiveCount = 0;
+    IngestedCount = 0;
+    FailureCount = 0;
+    ErrorMessage = "";
+  }
+
+  void onConfigIngested(const absl::string_view,
+                        const std::vector<Config::DecodedResourcePtr>&) override {
+    ++IngestedCount;
+  }
+
+  void onConfigIngested(const absl::string_view, const std::vector<Config::DecodedResourcePtr>&,
+                        const Protobuf::RepeatedPtrField<std::string>&) override {
+    ++IngestedCount;
+  }
+
+  void onConfigReceivedOrFailed(const envoy::service::discovery::v3::DiscoveryResponse&,
+                                const Config::ProcessingDetails& process_details) override {
+    countState(process_details.state_);
+    if (process_details.state_ == Config::ProcessingState::FAILED) {
+      ErrorMessage = process_details.error_detail_.message();
+    }
+  }
+
+  void onConfigReceivedOrFailed(const envoy::service::discovery::v3::DeltaDiscoveryResponse&,
+                                const Config::ProcessingDetails& process_details) override {
+    countState(process_details.state_);
+    if (process_details.state_ == Config::ProcessingState::FAILED) {
+      ErrorMessage = process_details.error_detail_.message();
+    }
+  }
+
+  static std::atomic<int> ReceiveCount;
+  static std::atomic<int> IngestedCount;
+  static std::atomic<int> FailureCount;
+  static std::string ErrorMessage;
+
+private:
+  void countState(const Config::ProcessingState& state) {
+    switch (state) {
+    case Config::ProcessingState::RECEIVED:
+      ++ReceiveCount;
+      break;
+    case Config::ProcessingState::FAILED:
+      ++FailureCount;
+      break;
+    };
+  }
+};
+
+std::atomic<int> TestXdsConfigTracker::ReceiveCount;
+std::atomic<int> TestXdsConfigTracker::IngestedCount;
+std::atomic<int> TestXdsConfigTracker::FailureCount;
+std::string TestXdsConfigTracker::ErrorMessage;
+
+static std::map<std::string, envoy::service::discovery::v3::Resource> ResourcesMap;
+
+class TestXdsConfigTrackerFactory : public Config::XdsConfigTrackerFactory {
+public:
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<test::envoy::config::xds::TestXdsConfigTracker>();
+  }
+
+  std::string name() const override { return "envoy.config.xds.test_xds_tracer"; };
+
+  Config::XdsConfigTrackerPtr createXdsConfigTracker(const ProtobufWkt::Any&,
+                                                     ProtobufMessage::ValidationVisitor&) override {
+    return std::make_unique<TestXdsConfigTracker>();
+  }
+};
+
+class XdsConfigTrackerIntegrationTest : public Grpc::DeltaSotwIntegrationParamTest,
```

Comment:
Please also test this against the unified mux

Reply:
Done for the unified mux.